modules/extension/wmts/src/main/java/org/geotools/ows/wmts/response/GetTileResponse.java

```diff
@@ -16,23 +16,51 @@
  */
 package org.geotools.ows.wmts.response;
 
+import java.awt.image.BufferedImage;
 import java.io.IOException;
 import java.net.URL;
 import org.geotools.data.ows.Response;
 import org.geotools.http.HTTPResponse;
+import org.geotools.image.io.ImageIOExt;
 import org.geotools.ows.ServiceException;
 import org.geotools.ows.wmts.model.WMTSServiceType;
 
-/** @author ian */
+/**
+ * Represents the response of a tile request.
+ *
+ * <p>In most cases this is a image, but it can also be a file like kml with reference to an image.
```

Comment:
Vector tiles are pretty common these days, they can also be returned, in that case, there is no image at all.